feat: unify structure and add types 5.005, 6.010 and 20.102

dmnkdmnt opened the pull request:
Moved remaining type implementations and tests from types.go to separate files.

Added types:
- 5.005: Ratio (0..255)
- 6.010: counter pulses (-128..127)
- 20.102: HVAC Mode

Fixed issue in validation test of DPT 10.001.

mobilarte commented:
The implementation of DPT 19.001 is slightly more complicated as the permissible hour range is [0...24] and not [0...23] and some combinations are ruled out (see KNX Datapoint Types).

dmnkdmnt commented:
I completely forgot about that o.O
Added that code for private use quite some time ago and this morning I asked myself why I never pushed it to GitHub and created a pull request.. I guess that's why, good catch! :)
I removed DPT 19.001 from the PR for now. Maybe I find some time in the future to complete the implementation.

vapourismo commented on knx/dpt/types_17.go:
```diff
@@ -0,0 +1,38 @@
+		return err
+	}
+
+	if *d <= 63 {
```
This must compare against `value` not `d`.

dmnkdmnt replied:
I see you already fixed this on master, I will bring that change to the PR later. :)

vapourismo commented on knx/dpt/types_18.go:
```diff
@@ -0,0 +1,40 @@
+		return err
+	}
+
+	if *d <= 63 || (*d >= 128 && *d <= 191) {
```
This also compares against `d` and should not do that.

dmnkdmnt replied:
Same for this.

vapourismo commented on knx/dpt/types_20.go:
```diff
@@ -0,0 +1,33 @@
+package dpt
+
+// DPT_20102 represents DPT 20.102 / HVAC Mode.
+type DPT_20102 uint8
```
Please add named constants for all values of this enum.

Like so:

```go
const (
  Auto DPT_20102 = 0
  Comfort = 1
  // ...
)
```

dmnkdmnt replied:
Done

vapourismo commented on knx/dpt/types_20.go:
```diff
@@ -0,0 +1,33 @@
+	case 4:
+		return "Building Protection"
+	default:
+		return ""
```
This is actually "reserved".

dmnkdmnt replied:
In the KNX master file of my test project this was not named, but ETS gave me an empty string.
What source du you usually use?

vapourismo replied:
It is defined in the KNX standard.

![image](https://user-images.githubusercontent.com/2806669/187203792-854c7d15-60f3-4bd4-b802-1ca78bd0226e.jpeg)

dmnkdmnt replied:
True, you find this in the types definition a lot in various versions. There is for example `reserved`, `reserved, unused`, `not used, reserved`, but in my understanding this is all referring to an unused value (-range) and not the value, or translation of these. Implementations, like the ETS, does not take them into account as well.
I would suggest to not translate this, but in the end it's your project, if you want to have it in I will change it. Just leave me your thoughts on this. :)

vapourismo replied:
The problem is that the empty string doesn't mean anything. Somebody encountering that who is not familiar with this discussion would assume something is wrong. Hence there must be some meaning conveyed. I am not too fixed on "reserved" - for all I care it could be "unspecified", "unknown" or similar. However, the empty string is simply not an acceptable choice.

dmnkdmnt replied:
I see your point, sounds good!
Maybe "unknown" is the best bet to go for when using it throughout the project.

Not said so far - thanks for all your work here, really appreciate it! :)

mobilarte replied:
I would prefer "reserved", since it is what the spec. says. "unknown" may imply it is unknown on the receiver side, but maybe not on the sender side, but we know that the value is reserved and currently not used.

dmnkdmnt replied:
Changed

vapourismo reviewed (changes requested):
Thanks for the contribution :heart: 

There are just a few things to address that have been changed recently  in `master` and have diverged in this PR.

A request for next time: When you move parts to a new file, make that a separate commit. Otherwise it is extremely hard to ensure consistency. Additionally, it is much harder to review if everything is 1-commit-1-pull-request.
